dll/win32/kernel32/client/compname.c

```diff
@@ -451,48 +459,90 @@ GetComputerNameW(LPWSTR lpBuffer, LPDWORD lpnSize)
     return ret;
 }
 
+static BOOL
+BaseVerifyDnsName(LPCWSTR lpDnsName)
+{
+    HINSTANCE hDNSAPI;
+    FN_DnsValidateName_W fnValidate;
+    NTSTATUS Status;
+    BOOL ret = FALSE;
+
+    hDNSAPI = LoadLibraryW(L"dnsapi.dll");
+    if (hDNSAPI == NULL)
+        return FALSE;
+
+    fnValidate = (FN_DnsValidateName_W)GetProcAddress(hDNSAPI, "DnsValidateName_W");
+    if (fnValidate)
+    {
+        Status = (*fnValidate)(lpDnsName, DnsNameHostnameLabel);
+        if (Status == STATUS_SUCCESS || Status == DNS_ERROR_NON_RFC_NAME)
+            ret = TRUE;
+    }
+
+    FreeLibrary(hDNSAPI);
+
+    return ret;
+}
 
 /*
  * @implemented
  */
-static
-BOOL
+static BOOL
 IsValidComputerName(COMPUTER_NAME_FORMAT NameType,
                     LPCWSTR lpComputerName)
 {
-    PWCHAR p;
-    ULONG Length;
+    BOOL ret;
+    SIZE_T Length;
+    static const WCHAR s_szInvalidChars[] =
+        L"\"/\\[]:|<>+=;,?"
+        L"\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F"
+        L"\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F";
 
-    /* FIXME: do verification according to NameType */
+    if (lpComputerName == NULL)
+        return FALSE;
 
-    Length = 0;
-    p = (PWCHAR)lpComputerName;
+    Length = wcslen(lpComputerName);
```

Comment:
The string is passed from user mode unchanged, need to use safe string functions here

Reply:
`RtlStringCchLengthW(lpComputerName, MAX_PATH, &Length);`

Reply:
```c
    /* get string length */
    if (!NT_SUCCESS(RtlStringCchLengthW(lpComputerName, NTSTRSAFE_MAX_CCH, &Length)))
        return FALSE;
```

Reply:
So yes, what is the actual advantage of using RtlStringCchLengthW over wcslen ?

Reply:
Maybe that is "buffer overrun". But I didn't understand the danger of it yet.

Reply:
I cannot imagine a wide string longer than `NTSTRSAFE_MAX_CCH`. The other value than `NTSTRSAFE_MAX_CCH` is better.
Cc @Extravert-ir

Reply:
That's 64. OK.